packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -132,3 +110,41 @@ export const SelectedItemCheck = styled( Ariakit.SelectItemCheck )`
 	margin-inline-start: ${ ITEM_PADDING };
 	font-size: 24px; // Size of checkmark icon
 `;
+
+const truncateStyles = css`
+	overflow: hidden;
+	text-overflow: ellipsis;
+	white-space: nowrap;
+`;
+
+export const ItemHintWrapper = styled.div`
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	flex-wrap: wrap;
+	flex: 1;
+`;
+
+export const SelectedExperimentalHintWrapper = styled.div`
+	${ truncateStyles }
+`;
+
+export const SelectedExperimentalHintItem = styled.span`
+	color: ${ COLORS.theme.gray[ 600 ] };
+	margin-inline-start: ${ space( 2 ) };
+`;
+
+export const ItemHintContent = styled.span`
+	padding-inline-end: ${ space( 4 ) };
+`;
+
+export const ItemHint = styled.span`
+	// Necessary to override V1 styles passed via the legacy
+	// '.components-custom-select-control__item-hint' class
+	${ SelectItem } && {
+		color: ${ COLORS.theme.gray[ 600 ] };
+		text-align: initial;
+		line-height: 1.4;
+		padding-inline-end: ${ space( 1 ) };
```

Comment:
Using padding here to better match legacy behavior. Also, the V1 styles are loaded in the editor, if we use the legacy `components-custom-select-control__item` we will get those styles applied to the V2 as well 😅 Without changing `margin-inline-end` to `padding-inline-end`, the hint would get both a margin and a padding applied